gensim/test/test_fasttext.py

```diff
@@ -1300,6 +1306,253 @@ def _run(self, fin):
+            self.assertEqual(set(model_trained.wv.index2word), set(model_loaded.wv.index2word))
+
+            for w in model_trained.wv.index2word:
+                v_orig = model_trained.wv[w]
+                v_loaded = model_loaded.wv[w]
+
+                try:
+                    self.assertLess(calc_max_diff(v_orig, v_loaded), MAX_WORDVEC_COMPONENT_DIFFERENCE)
+                except AssertionError as e:
+                    s = e.args[0]
+                    s += " (max difference for components in wordvector of \"%s\"" % w
+                    s += " larger than the thershold %f" % MAX_WORDVEC_COMPONENT_DIFFERENCE
+                    e.args = (s,)
+                    raise e
+
+    def test_round_trip_model_model_skipgram(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 1,
+            "negative": 5,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_model_model(model_params)
+
+    def test_round_trip_model_model_cbow(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 0,
+            "negative": 5,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_model_model(model_params)
+
+
+def _read_binary_file(fname):
+    with open(fname, "rb") as f:
+        data = f.read()
+    return data
+
+
+class SaveFacebookFormatFileGensimTest(unittest.TestCase):
+    """
+    This class containts tests that check the following scenario:
+
+    + create binary fastText file model1.bin using gensim
+    + load file model1.bin to variable `model`
+    + save `model` to model2.bin
+    + check if files model1.bin and model2.bin are byte identical
+    """
+
+    def _check_roundtrip_file_file(self, model_params):
+        with temporary_file("roundtrip_file_to_file1.bin") as fpath1, \
+            temporary_file("roundtrip_file_to_file2.bin") as fpath2:
+            _create_and_save_test_model(fpath1, model_params)
+            model = gensim.models.fasttext.load_facebook_model(fpath1)
+            gensim.models.fasttext.save_facebook_model(model, fpath2)
+            self.assertEqual(_read_binary_file(fpath1), _read_binary_file(fpath2))
+
+    def test_roundtrip_file_file_skipgram(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 1,
+            "negative": 0,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_file_file(model_params)
+
+    def test_roundtrip_file_file_cbow(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 0,
+            "negative": 0,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_file_file(model_params)
+
+
+def _save_test_model(out_base_fname, model_params, fasttext_cmd):
```

Comment:
I think the name is misleading here, because you're not just saving, you're _training_ a model here.

But really, I think this function may be unnecessary. Just train the two models locally, and add the binaries to the git tree. How large could they be?

Reply:
Creating the model explicitly can be useful. 

Imagine Facebook changes the format.  Once we provide the latest FT_CMD this will cause test failure. Once we correct our implementation tests will pass again.  

If we include binary files we live in ignorance, and we have to regenerate the model file for the test to work again.